onnx/defs/tensor/defs.cc

```diff
@@ -836,6 +836,71 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
+static const char* ScatterND_ver11_doc = R"DOC(
+ScatterND takes three inputs `data` tensor of rank r >= 1, `indices` tensor of rank q >= 1,
+and `updates` tensor of rank q + r - indices.shape[-1] - 1. The output of the operation
+is produced by creating a copy of the input `data`, and then updating its value to values
+specified by `updates` at specific index positions specified by `indices`. Its output shape
+is the same as the shape of `data`.
+
+`indices` is an integer tensor. The last dimension of `indices` can be at most the rank of
+`data`:
+
+    indices.shape[-1] <= rank(data)
+
+`updates` is a tensor with shape
+
+    indices.shape[:-1] + data.shape[indices.shape[-1]:]
+
+The `output` is calculated via the following equation:
+
+    output = data
+    output[indices[(i_0, ..., i_{q-2})]] = updates[(i_0, ..., i_{q-2})]
```

Comment:
Can you add an iteration loop around the update equation (the second line)? I think we could use numpy's ndindex to express the iteration. Something like below
```python
   update_indices = indices.shape[:-1]
   for idx in np.ndindex(update_indices):
      output[indices[idx]] = updates[idx]
```
You may have to adjust a few things above to make it run, but as pseudo-code, hopefully it will be clear.